doc/nrf/libraries/others/audio_module.rst

```diff
@@ -0,0 +1,119 @@
+
+.. figure:: images/audio_module_functions.svg
+   :alt: Audio module functions
+
+
+The following table outlines the available functions that are defined in :c:struct:`audio_module_functions` and whether they are mandatory or not:
+
+.. list-table::
+    :header-rows: 1
+
+    * - Function
+      - Mandatory/Optional
+      - Comment
+    * - ``audio_module_functions.*open``
+      - Optional
+      - Perform any operations necessary to open the module implementation.
+    * - ``audio_module_functions.*close``
+      - Optional
+      - Tidy up and close the module implementation.
+    * - ``audio_module_functions.*configure_set``
+      - Mandatory
+      - Configure the module implementation to perform a particular operation.
+    * - ``audio_module_functions.*configuration_get``
+      - Mandatory
+      - Return the internal configuration of the module implementation.
+    * - ``audio_module_functions.*start``
+      - Optional
+      - Perform any operations necessary to set the module implementation running.
+    * - ``audio_module_functions.*stop``
+      - Optional
+      - Perform any operations to necessary stop the module implementation.
+    * - ``audio_module_functions.*data_process``
+      - Mandatory
+      - Process the data within the module implementation.
+
+An module implementation can run only if these user provided functions are defined and given to the audio module.
+The audio module framework itself cannot perform any tasks, as it merely supplies a consistent way to interface to an audio algorithm.
+
+The following figure show the internal states of the audio module:
+
+.. figure:: images/audio_module_states.svg
+   :alt: Audio module internal states
+
+Configuration
+*************
+
+To use the audio module library, set the following Kconfig options to ``y`` in the project configuration file :file:`prj.conf`:
+
+* :kconfig:option:`CONFIG_AUDIO_MODULE`
+* :kconfig:option:`CONFIG_DATA_FIFO`
+
+Application integration
+***********************
+
+To create your own audio module for an LE Audio application, complete the following steps:
+
+#. Write the mandatory functions required by the function table API in `Implementation`_.
+#. Write any optional functions.
+#. Assign the function table to an instance of an audio module.
+
+The audio application opens the module, configures it and connects it to other modules, the application or both.
+The module can then be started and you can transfer data in accordance to what type the module is :c:enum:`audio_module_type`.
+
+The following figure demonstrates a simple decoding audio system, where the decoded audio is sent to an I2S output and returned to the application:
+
+.. figure:: images/audio_module_example.svg
+   :alt: Audio module stream example
+
```

Comment:
"The process blob is an Input-Output module, while the output blob is an Output module."

Either include a sentence like this, or modify the figure.

Reply:
Updated the figure.